src/lang-promql/complete/hybrid.ts

```diff
@@ -370,10 +394,12 @@ export function analyzeCompletion(state: EditorState, node: SyntaxNode): Context
 export class HybridComplete implements CompleteStrategy {
   private readonly prometheusClient: PrometheusClient | undefined;
   private readonly maxMetricsMetadata: number;
+  private readonly enricher: EnrichCompletionHandler;
 
-  constructor(prometheusClient?: PrometheusClient, maxMetricsMetadata = 10000) {
+  constructor(prometheusClient?: PrometheusClient, maxMetricsMetadata = 10000, enricher = defaultEnricher) {
```

Comment:
Would passing an options struct become more natural at this point, with more and more parameters, all of which are optional? Otherwise if I want to specify an enricher, I also have to specify the first two parameters now.

Reply:
mmm yeah that's true. What name do you think we should use for this struct ?

Reply:
`interface HybridCompleteConfig`, and then just `config` here as the parameter name?

Reply:
mmm actually does it really matter ? In the current design you don't instantiate yourself the HybridComplete. You are passing through the config `CompletionConfig` and through the class `PromQLExtension`.

So after some reflexions, I don't really see the value here. It will create another internal struct / interface

Reply:
Ah yeah, didn't see that. As long as it's only used internally it's fine to keep it like this I think.